indexing-service/src/test/java/org/apache/druid/indexing/overlord/ImmutableWorkerInfoTest.java

```diff
@@ -225,11 +239,14 @@ public void test_canRunTask()
         new Worker("http", "testWorker2", "192.0.0.1", 10, "v1", WorkerConfig.DEFAULT_CATEGORY),
         6,
         0,
+        Collections.emptyMap(),
         ImmutableSet.of("grp1", "grp2"),
         ImmutableSet.of("task1", "task2"),
         DateTimes.of("2015-01-01T01:01:02Z")
     );
 
+    WorkerTaskRunnerConfig config = mock(WorkerTaskRunnerConfig.class);
```

Comment:
This will require changing the config class as currently we don't have setters or constructors (except NoArgs), to use actual config object the most painless approach is to add setters and mark them as VisibleForTesting, @kfaraz WDYT?

Reply:
No, you can also create an inline class like:

```
WorkerTaskRunnerConfig config = new WorkerTaskRunnerConfig() {
    @Override
    public Map<String, Double> getTaskSlotRatiosPerWorker() {
          // return your value here
    }
}
```

Later, we can add a constructor.

Reply:
Thanks for the hint, will do it this way

Reply:
Updated